Essentials/src/main/java/com/earth2me/essentials/MetaItemStack.java

```diff
@@ -226,28 +226,25 @@ public void addStringMeta(final CommandSource sender, final boolean allowUnsafe,
             final BookMeta meta = (BookMeta) stack.getItemMeta();
             meta.setTitle(title);
             stack.setItemMeta(meta);
-        } else if (split.length > 1 && split[0].equalsIgnoreCase("power") && MaterialUtil.isFirework(stack.getType()) && hasMetaPermission(sender, "firework-power", false, true, ess)) {
-            final int power = NumberUtil.isInt(split[1]) ? Integer.parseInt(split[1]) : 0;
-            final FireworkMeta meta = (FireworkMeta) stack.getItemMeta();
-            meta.setPower(power > 3 ? 4 : power);
-            stack.setItemMeta(meta);
         } else if (split.length > 1 && split[0].equalsIgnoreCase("itemflags") && hasMetaPermission(sender, "itemflags", false, true, ess)) {
             addItemFlags(string);
         } else if (MaterialUtil.isFirework(stack.getType())) {
-            //WARNING - Meta for fireworks will be ignored after this point.
-            addFireworkMeta(sender, false, string, ess);
+            if (split[0].equalsIgnoreCase("power") || !parseEnchantmentStrings(sender, allowUnsafe, split, ess)) {
```

Comment:
Why the check for `power`? It isn't required in the firework meta

Reply:
Moved lines 229-233 to addFireworkMeta method, so it is required now

Reply:
It shouldn't be though? We don't need the full meta to create a valid firework

Reply:
This change does not affect the creation of fireworks, but rather makes it more consistent - this is the only action that belongs to fireworks meta, but isn't included in the method.
I might be just stupid, but I really see no reason for it not to be there